python/triton_mlir/triton_mlir/TritonExtension.cpp

```diff
@@ -354,6 +363,61 @@ createTargetMachine(llvm::Module *module, std::string proc,
   return machine;
 }
 
+/// addPassesToX helper drives creation and initialization of TargetPassConfig.
+static llvm::TargetPassConfig *
+addPassesToGenerateCode(llvm::CodeGenTargetMachineImpl &TM,
+                        llvm::PassManagerBase &PM, bool DisableVerify,
+                        llvm::MachineModuleInfoWrapperPass &MMIWP) {
+  // Targets may override createPassConfig to provide a target-specific
+  // subclass.
+  llvm::TargetPassConfig *PassConfig = TM.createPassConfig(PM);
+  PassConfig->insertPass(&llvm::SIFoldOperandsLegacyID,
+                         &llvm::MaxsMachineFunctionID);
```

Comment:
insert my pass here into the "pass config" to run right after the `SIFoldOperandsLegacy` pass